app/src/main/java/com/ciandt/dragonfly/example/components/chips/ChipsView.kt

```diff
@@ -0,0 +1,140 @@
+package com.ciandt.dragonfly.example.components.chips
+
+import android.content.Context
+import android.content.res.Resources
+import android.graphics.drawable.Drawable
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.RelativeLayout
+import com.ciandt.dragonfly.example.R
+import kotlinx.android.synthetic.main.component_chips_view.view.*
+
+
+class ChipsView : RelativeLayout {
+
+    private val DEFAULT_EDGES_WIDTH = dpToPx(16)
+    private val DEFAULT_SPACE_BEFORE = dpToPx(0)
+    private val DEFAULT_SPACE_BETWEEN = dpToPx(0)
+    private val DEFAULT_SPACE_AFTER = dpToPx(0)
+
+    private val chips = ArrayList<Chip>()
+
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var adapter: ChipAdapter
+
+    private var selectCallback: ((Chip) -> Unit)? = null
+    private var deselectCallback: ((Chip) -> Unit)? = null
+
+    constructor(context: Context) : super(context) {
+        initializeViews(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+        initializeViews(context)
+        initializeAttributes(attrs)
+    }
+
+    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
+        initializeViews(context)
+        initializeAttributes(attrs)
+    }
+
+    fun initializeViews(context: Context) {
+        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        inflater.inflate(R.layout.component_chips_view, this)
+
+        adapter = ChipAdapter(context, chips) { chip ->
+            if (chip.isSelected()) {
+                selectCallback?.invoke(chip)
+            } else {
+                deselectCallback?.invoke(chip)
+            }
+        }
+        adapter.setHasStableIds(true)
+
+        recyclerView = findViewById(R.id.list) as RecyclerView
+        recyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+        recyclerView.adapter = adapter
+        recyclerView.hasFixedSize()
+    }
+
+    fun initializeAttributes(attrs: AttributeSet) {
+
+        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.ChipsView, 0, 0)
+        try {
+
+            val chipLayout = typedArray.getResourceId(R.styleable.ChipsView_chipLayout, -1)
+            if (chipLayout != -1) {
```

Comment:
Is -1 the UNINITIALIZED_LAYOUT_RESOURCE proposed above? If so, I think it's clear with a const.